src/test/java/com/eleks/academy/whoami/service/GameServiceImplTest.java

```diff
@@ -0,0 +1,56 @@
+package com.eleks.academy.whoami.service;
+
+import com.eleks.academy.whoami.core.SynchronousGame;
+import com.eleks.academy.whoami.core.impl.PersistentGame;
+import com.eleks.academy.whoami.model.request.NewGameRequest;
+import com.eleks.academy.whoami.repository.GameRepository;
+import com.eleks.academy.whoami.service.impl.GameServiceImpl;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.times;
+
+@ExtendWith(MockitoExtension.class)
+public class GameServiceImplTest {
+
+	@Mock
+	private GameRepository gameRepository;
+
+	@InjectMocks
+	private GameServiceImpl gameService;
+
+	private final NewGameRequest gameRequest = new NewGameRequest();
+
+	@BeforeEach
+	public void setMockMvc() {
+		gameRequest.setMaxPlayers(4);
+	}
+
+	@Test
+	void createGame() {
+		final String player = "player";
+		SynchronousGame game = new PersistentGame(player, gameRequest.getMaxPlayers());
+
+		when(gameRepository.save(any(SynchronousGame.class))).thenReturn(game);
+
+		var createdGame = gameService.createGame(player, gameRequest);
+
+		assertNotNull(createdGame);
```

Comment:
here you have checks(assertions) that value if not null, but much better to compare the result as an object with the object of the same type and values used for comparison. Library assertJ(should be already included) has the method `assertThat()` using it we can compare results with expectation in following way:
```
assertThat(gameService.createGame(player, gameRequest)).isEqualTo(new PersistentGame("player", 4))
```

Reply:
Resolved
But with a few changes. I need the same type.